hal/src/nRF52840/sleep_hal.cpp

```diff
@@ -590,6 +602,22 @@ static int enterStopBasedSleep(const hal_sleep_config_t* config, hal_wakeup_sour
         // we are still under the effect of HAL_disable_irq that masked all but SoftDevice interrupts using BASEPRI
         __enable_irq();
 
+        /*
+         * SD_EVT_IRQn is set pending after __enable_irq() and SoftDevice interrupts being executed.
```

Comment:
Let's reword this, I had to read it multiple times and ask additional questions to understand how this works:

Wake-up on BLE events relies mainly on `SD_EVT_IRQn` interrupt, which will be disabled through NVIC (not masked) when entering `sd_nvic_critical_region_enter()`. Normally on BLE activity we would be woken up by other high-priority SoftDevice interrupts (e.g. timers, radio events etc), however `SD_EVT_IRQn` would be set as pending only after we unmask high-priority SoftDevice interrupts (`__enable_irq()`) and after the appropriate SoftDevice interrupt handlers finish executing. We would then immediately go back to sleep and wouldn't notice this change in `SD_EVT_IRQn` state. As a workaround, for the duration of the sleep, we'll be bumping the priority of `SD_EVT_IRQn`, so that it can wake us up from stop mode (`WFI`) as well.

Reply:
Alternatively, I think it might be viable to simply move pending IRQ checks under `__enable_irq()`.